koku/api/settings/tags/mapping/view.py

```diff
@@ -85,9 +85,10 @@ class SettingsTagMappingView(generics.GenericAPIView):
     def get(self, request: Request, **kwargs):
         filtered_qset = self.filter_queryset(self.get_queryset())
         serializer = self.serializer_class(filtered_qset, many=True)
-        paginator = ListPaginator(serializer.data, request)
+        formatted_response = format_tag_mapping_relationship(serializer)
```

Comment:
Ther are no type hints on `format_tag_mapping_relationship` so I'm confused. This looks like the caller went from passing in a `Response` object to a `Serializer` object even though the parameter name is `original_response`. That seems to work since they both have a `data` property/attribute, but it may make the interface of `format_tag_mapping_relationship` more clear if it accepts a MutableMapping and returns a MutableMapping.
```suggestion
        formatted_response = format_tag_mapping_relationship(serializer.data)
```

```python
def format_tag_mapping_relationship(data: dict[str: dict[str:str]]) -> dict[str: dict[str:str]]:
   ....
```

Even better would be to use a Pydantic model or dataclass to enforce the type and shape of the data.

Reply:
Here's a rough concept of what storing this data in objects could look like.
```python
import dataclasses

@dataclasses.dataclass(frozen=True)
class TagKey:
    uuid: str
    key: str
    source_type: str
    

@dataclasses.dataclass(frozen=True)
class Relationship:
    parent: TagKey
    child: TagKey
    children: list[TagKey] = dataclasses.field(default_factory=list)
```